source/exe/main_common.cc

```diff
@@ -81,6 +84,28 @@ MainCommonBase::MainCommonBase(OptionsImpl& options) : options_(options) {
 
 MainCommonBase::~MainCommonBase() { ares_library_cleanup(); }
 
+void MainCommonBase::configureComponentLogLevels() {
+  const std::vector<std::pair<std::string, std::string>>& component_log_levels =
+      options_.componentLogLevels();
+  for (auto& component_log_level : component_log_levels) {
+    const std::string& component = component_log_level.first;
+    const std::string& log_level = component_log_level.second;
+    size_t level_to_use = std::numeric_limits<size_t>::max();
```

Comment:
Shouldn't we check if this is never actually set because the log level is invalid? I suspect you can make more of this function into a utility function so that it can be more easily tested?

Reply:
Nevermind I see that you do check this. Can you just have the function return a pair of the int level and the logger reference? Do we need to do this search again? At the very least can you ASSERT we have valid values here?

Reply:
I have made the following changes
- Made componentLogLevels() in OptionsImpl return pair of log_name,int so that we do not recompute the log level again here. I tried to get Logger reference in pair but it is difficult as Logger is not available in /include to use in options.h. We could forward declare it but I think this is better
- Moved the method to get Logger given log name to Registry so that it can be used else where. I have added tests for it.
- Added ASSERT here for additional safety.